### PR TITLE
[dv/clkmgr] Update testplan

### PR DESCRIPTION
**Pull request by matutem**
Update the testplan and DV doc with jitter test.
Add extclk sequence description to DV doc.
Enhance some test descriptions in testplan.
Add planned covergroups in testplan.
Removes mention of the interrupt functionality.
Add planned SVA assertions for various checks predicated on scanmode
being inactive.

Signed-off-by: Guillermo Maturana <maturana@google.com>

### CODE REVIEW

**Comment by matutem** [comment]
I think these are the pending items from the V1 review.

**Review by weicaiyang — APPROVED** [review]
LGTM

File: hw/ip/clkmgr/data/clkmgr_testplan.hjson [hunk 1]
```diff
@@ -139,5 +182,21 @@
...
+            and the `lc_dft_en_i`, `lc_clk_byp_req_i`, and `scanmode_i`
+            input pins. This covergroup collects their cross.
+            '''
+    }
+    {
+      name: jitter_cg
+      desc: '''
+            Collects coverage of the `jitter` CSR.
+            '''
+    }
```
Comment (sriyerg) [inline hw/ip/clkmgr/data/clkmgr_testplan.hjson:200]: This should show up as toggle on clkmgr IOs, right?

Comment (matutem) [reply hw/ip/clkmgr/data/clkmgr_testplan.hjson:200]: This intends to collect coverage on the CSR. The jitter_en_o output pin will show up in toggle coverage, and the test checks their causal relation. Seems a bit redundant, and I would not mind removing this.

By the way, do we automagically collect coverage on all CSRs?

Comment (sriyerg) [reply hw/ip/clkmgr/data/clkmgr_testplan.hjson:200]: Okay, I'm fine either way.

File: hw/ip/clkmgr/data/clkmgr_testplan.hjson [hunk 2]
```diff
@@ -98,14 +107,48 @@
...
             **Checks**:
-            - SVA assertions:
-              - `lc_clk_byp_req_i == lc_ctrl_pkg::On` causes
-                `ast_clk_byp_req_o == lc_ctrl_pkg::On`.
-              - `extclk_sel == lc_ctrl_pkg::On` and
-                `lc_dft_en_i == lc_ctrl_pkg::On` causes
-                `ast_clk_byp_req_o == lc_ctrl_pkg::On`.
-              - `ast_clk_byp_ack_i == lc_ctrl_pkg::On` causes the divided
-                clocks to ramp up unless `scanmode_i == lc_ctrl_pkg::On`.
+            Clock divider checks: since `scanmode_i` is asynchronous the checks
```
Comment (sriyerg) [inline hw/ip/clkmgr/data/clkmgr_testplan.hjson:110]:
```suggestion
            Clock divider checks: since `scanmode_i` is asynchronous, the checks
```

**Review by sriyerg — APPROVED** [review]

**Pull request merged**